tet.rs/src/tetris/board/my_board.rs

```diff
@@ -0,0 +1,382 @@
+    active_piece_position: PiecePosition,
+    queue: QueueT,
+    second_bag: QueueT,
+    held_piece: HeldPieceT,
+    has_held: bool,
+    rng: WyRand,
+}
+
+impl std::fmt::Display for Board {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self.held_piece {
+            Some(piece) => write!(f, "[ {} ]", piece),
+            None => write!(f, "[   ]"),
+        }?;
+
+        write!(f, " ")?;
+
+        for i in 0..QUEUE_DISPLAY_SIZE {
+            write!(
+                f,
+                "{}{}",
+                self.get_queue_at(i),
+                if i + 1 < QUEUE_DISPLAY_SIZE {
+                    " "
+                } else {
+                    "\n"
+                }
+            )?;
+        }
+
+        for y in (0..(BOARD_HEIGHT + 4)).rev() {
+            write!(f, "{}", PieceRepresentation::get_border())?;
+
+            for x in 0..BOARD_WIDTH {
+                write!(
+                    f,
+                    "{}",
+                    if self.get_cell(x, y) {
+                        PieceRepresentation::get_taken()
+                    } else {
+                        PieceRepresentation::get_empty()
+                    }
+                )?;
+            }
+
+            writeln!(f, "{}", PieceRepresentation::get_border())?;
+        }
+
+        for _ in 0..BOARD_WIDTH + 2 {
+            write!(f, "{}", PieceRepresentation::get_border())?;
+        }
+
+        writeln!(f)?;
+
+        Ok(())
+    }
+}
+
+impl Board {
+    pub fn new() -> Self {
+        let mut board: Self = Self {
+            board: BoardT::from_elem((BOARD_WIDTH * BOARD_HEIGHT * 2) as usize, false),
+            active_piece_type: PieceType::TI, // We'll set a new one right after
+            active_piece_position: PiecePosition::new(0, 0, PieceRotation::RN), // Same, this will get a new value
+            queue: QueueT::new(),
+            second_bag: QueueT::new(),
+            held_piece: None,
+            has_held: false,
+            rng: WyRand::new(),
+        };
+
+        board.ensure_complete_queue(true);
+        board.spawn_next_piece();
+
+        board
+    }
+
+    pub fn get_cell(&self, x: PositionT, y: PositionT) -> bool {
+        let index: usize = Board::xy_to_index(x, y);
+        self.board[index]
+    }
+
+    pub fn set_cell(&mut self, x: PositionT, y: PositionT, val: bool) {
+        let index: usize = Board::xy_to_index(x, y);
+        self.board.set(index, val);
+    }
+
+    fn get_random_bag(&mut self) -> QueueT {
+        let mut bag = [
+            PieceType::TI,
+            PieceType::TO,
+            PieceType::TJ,
+            PieceType::TL,
+            PieceType::TS,
+            PieceType::TZ,
+            PieceType::TT,
+        ];
+        self.rng.shuffle(&mut bag);
+
+        let mut resulting_bag = QueueT::with_capacity(bag.len());
+        for piece in bag.iter() {
+            resulting_bag.push_back(*piece);
+        }
+
+        resulting_bag
```

Comment:
```suggestion
        bag.iter().copied().collect()
```
There might be a more concise way of expressing this, but into_iter only returns a reference to PieceType for some reason.

Reply:
Oh thank you for that, it did seem like a horrible piece of code from my end.